binding.gyp

```diff
@@ -134,12 +134,6 @@
                 ],
                 'GCC_ENABLE_CPP_EXCEPTIONS': 'YES',
                 'GCC_ENABLE_EXCEPTIONS': 'YES',
-                'OTHER_CFLAGS': [
-                  "<!(echo \"-arch ${ARCH:=x86_64}\")",
-                ],
-                'OTHER_LDFLAGS': [
-                  "<!(echo \"-arch ${ARCH:=x86_64}\")",
-                ]
```

Comment:
The gyp flags are used for ZeroMQ, and build.js is used for libzmq (the upstream library used by ZeroMQ). There's a subtle difference between these two that's sometimes confusion for new contributors.

Reply:
definitely confusing to me! Where does the `ARCH` value here supposedly come from? I was finding it to be blank.

Reply:
ARCH is defined in the CI or as an env variable in `build.ts`

Reply:
If `ARCH` is always defined, there should be no need for `"<!(echo \"-arch ${ARCH:=x86_64}\")",`. It seems odd too that we should have to override the flags instead of simply omitting them.

I'll create an issue for what I thought was a symptom of this.